### PR TITLE
ENG-17 oxd-quick-search improvements

### PR DESCRIPTION
**Pull request by thilinasan16**
This commit contains improvements of oxd-quick-search

Update unit tests
Storybook update
read-only,focus,disabled missing styles update according to oxd.
## Checklist

- [x] Test Coverage is 100% for the newly added code
- [ ] Storybook stories are added/updated for the changed areas
- [x] Components standards defined [in this document](https://docs.google.com/document/d/16_Nd3VxE_lTD9pVkONQ0egn7IiwyX1pVXZjzl-V4tU8/) are followed
- [ ] Code is linted properly
- [x] Developer testing is done for the affected areas
- [ ] Package version updated (not applicable to ent branch)
- [ ] Changelog.md updated on possible breaking (applicable to ent branch)

### CODE REVIEW

File: storybook/stories/core/components/Input/QuickSearch.stories.js [hunk 1]
```diff
@@ -1,21 +1,80 @@
...
+      defaultValue: [],
+      table: {
+        type: {summary: 'emit event when dropdown opened'},
+      },
+    },
+    'dropdown:closed': {
+      control: {type: 'array'},
+      defaultValue: [],
+      table: {
+        type: {summary: 'emit event when dropdown closed'},
```
Comment (lgtkaushalya) [inline storybook/stories/core/components/Input/QuickSearch.stories.js:35]: First letter should be capital. Make sure other places also has the same

File: components/src/core/components/Input/__tests__/quick-search-input.spec.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      },
+    });
+    const input = wrapper.findComponent(AutocompleteInput);
+    input.setData({dropdownOpen: true});
+    input.vm.onCloseDropdown();
+    await wrapper.vm.$nextTick();
+    expect(wrapper.emitted('dropdown:closed')).toBeTruthy();
+  });
+
+  it('should not able to remove selected chip when readonly mode and  multi select mode', async () => {
```
Comment (lgtkaushalya) [inline components/src/core/components/Input/__tests__/quick-search-input.spec.ts:131]: Do we have multi select in Quick search?

File: components/src/core/components/Input/Autocomplete/AutocompleteInput.vue [hunk 3]
```diff
@@ -94,7 +94,6 @@ export default defineComponent({
   emits: [
     'update:modelValue',
     'dropdown:clear',
-    'dropdown:opened',
```
[the remark this answers is not in the document]
Comment (thilinasan16) [reply components/src/core/components/Input/Autocomplete/AutocompleteInput.vue:97]: As it has not been used in autocomplete

File: components/src/core/components/Input/__tests__/quick-search-input.spec.ts (hunk 2 above)
Comment (thilinasan16) [reply components/src/core/components/Input/__tests__/quick-search-input.spec.ts:131]: yes. same inherited behaviour of autocomplete

File: components/src/core/components/Input/__tests__/quick-search-input.spec.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+          label: 'HR Admin',
+        },
+      },
+    });
+    wrapper.find('[data-test="autocompleteSelectClearIcon"]').trigger('click');
+    await wrapper.vm.$nextTick();
+    expect(wrapper.emitted('update:modelValue')).toEqual([[null]]);
+  });
+
+  it('on AutoComplete component emits "select enter"  will emit "select enter"', async () => {
```
[the remark this answers is not in the document]
Comment (thilinasan16) [reply components/src/core/components/Input/__tests__/quick-search-input.spec.ts:98]: updated

File: storybook/stories/core/components/Input/QuickSearch.stories.js [hunk 5]
```diff
@@ -1,21 +1,80 @@
...
+      defaultValue: [],
+      table: {
+        type: {summary: 'emit event when select value updates'},
+      },
+    },
+    default: {
+      control: {type: 'object'},
+      defaultValue: [],
+      table: {
+        type: {summary: 'Default slots for autocomplete'},
```
[the remark this answers is not in the document]
Comment (thilinasan16) [reply storybook/stories/core/components/Input/QuickSearch.stories.js:63]: updated

File: components/src/core/components/Input/Autocomplete/QuickSearchInput.vue [hunk 6]
```diff
@@ -31,15 +32,31 @@ export default defineComponent({
...
+      default: false,
+    },
+    readonly: {
+      type: Boolean,
+      default: false,
+    },
+  },
   emits: [
     'update:modelValue',
     'update:searchTerm',
```
[the remark this answers is not in the document]
Comment (thilinasan16) [reply components/src/core/components/Input/Autocomplete/QuickSearchInput.vue:47]: updated

File: storybook/stories/core/components/Input/QuickSearch.stories.js (hunk 1 above)
Comment (thilinasan16) [reply storybook/stories/core/components/Input/QuickSearch.stories.js:35]: updated

**Pull request merged**